smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/WaiterGenerator.java

```diff
@@ -81,16 +83,34 @@ private void generateWaiter() {
                             waiter.getMaxDelay());
             writer.write("return createWaiter({...serviceDefaults, ...params}, input, checkState);");
         });
+
+        // generates WaitUtil....
+        writer.writeDocs(waiter.getDocumentation().orElse("") + " \n"
+                + " @param params : Waiter configuration options.\n"
+                + " @param input : the input to " + operationSymbol.getName() + " for polling.");
+        writer.openBlock("export const waitUntil$L = async (params: WaiterConfiguration<$T>, input: $T): "
+                + "Promise<WaiterResult> => {", "}", waiterName, serviceSymbol, inputSymbol, () -> {
+            writer.write("const serviceDefaults = { minDelay: $L, maxDelay: $L };", waiter.getMinDelay(),
+                    waiter.getMaxDelay());
+            writer.write("const result = await createWaiter({...serviceDefaults, ...params}, input, checkState);");
```

Comment:
Call waitFor here?

Reply:
Nah, since waitFor is marked deprecated, if we call it here then it gets "crossed out" in the IDE. We shouldn't use it. I dont think thats a great experience.

On the other hand, copy-pasting code isnt great either - even if its only 2 lines. On the whole I think its better if we dont call the deprecated function